packages/workflow-core/src/lib/logger.ts

```diff
@@ -0,0 +1,21 @@
+import { AnyRecord } from '@ballerine/common';
+
+export type LoggerInterface = {
+  log(message: string, logData?: AnyRecord): void;
+  error(message: string, logData?: AnyRecord): void;
+  warn(message: string, logData?: AnyRecord): void;
+  debug(message: string, logData?: AnyRecord): void;
+};
+
+/* eslint-disable no-console */
+export let logger: LoggerInterface = {
+  log: (message: string, logData: AnyRecord = {}) => console.log(message, logData),
+  error: (message: string, logData: AnyRecord = {}) => console.error(message, logData),
+  warn: (message: string, logData: AnyRecord = {}) => console.warn(message, logData),
+  debug: (message: string, logData: AnyRecord = {}) => console.debug(message, logData),
+};
+/* eslint-enable no-console */
+
+export const setLogger = (newLogger: LoggerInterface) => {
+  logger = newLogger;
+};
```

Comment:
I decided to make the logger a singleton in the WF-core, as it is already a singleton in the WF-service. It's easier than passing the logger when calling `createWorkflow` and then drilling it down to all other functions.